libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt

```diff
@@ -180,16 +185,19 @@ internal object MapboxNavigationTelemetry {
         when (sessionState) {
             IDLE, FREE_DRIVE -> {
                 sessionStop()
+                eventsSessionId = obtainUniversalUniqueIdentifier()
                 handleStateChanged(this.sessionState, sessionState)
             }
             ACTIVE_GUIDANCE -> {
                 locationsCollector.flushBuffers()
-                handleStateChanged(this.sessionState, sessionState)
                 needStartSession = true
+                handleStateChanged(this.sessionState, sessionState)
+                eventsSessionId = obtainUniversalUniqueIdentifier()
                 startSessionIfNeedAndCan()
             }
         }
         this.sessionState = sessionState
+        eventsSessionIdObserver?.onEventsSessionIdChanged(eventsSessionId)
```

Comment:
Is it possible for the state to change multiple times for the same navigation session?

For example, does the sessionId change for every route change? Does this mean re-routes create new sessions? It looks like this observer is triggered whenever the state is _updated_, not just when it _changes_

https://github.com/mapbox/mapbox-navigation-android/blob/ca57002edb30b82ed2cb9c0eb3180d6beda951fa/libnavigation-core/src/main/java/com/mapbox/navigation/core/NavigationSession.kt#L20